addons/account/models/account_move_line.py

```diff
@@ -3389,6 +3389,16 @@ def _prepare_edi_vals_to_export(self):
         }
         return res
 
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        # Add total amount_currency to the read_group when view is grouped by currency_id only
+        res = super().read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
+        if 'currency_id' in groupby:
+            for group_line in res:
+                lines = self.search(group_line.get('__domain'))
```

Comment:
Just so you know, doing a search inside of a loop is very inefficient and should be avoided. You can most of the time replace it with a read_group before the loop in order to do only a single query to the database.